demo/demo/tests/test_paddle.py

```diff
@@ -17,6 +20,141 @@
 from subscriptions.utils import fromisoformat
 
 
+def automate_payment(url, card):
+    """
+    This function replicates all the steps that are done at client level in the paddle payment page.
+    """
+    email = settings.PADDLE_TEST_EMAIL
+    assert email, 'Please configure the `PADDLE_TEST_EMAIL` environment variable.'
+    # get the browser info token, which will be used later on.
+    browser_info = url.strip('/').split('/')[-1]
+    session = requests.Session()
+
+    # hit the redirect url, and start the payment procedure
+    payment_page = session.get(url, allow_redirects=True)
+
+    # the checkout id has been initialized.
+    payment_page_query = parse.parse_qs(parse.urlsplit(payment_page.url).query)
+    checkout_id = payment_page_query['checkout_id'][0]
+    assert 'checkout_id' in payment_page_query
+
+    # parse the react to get the needed urls
+    checkout_api_url = payment_page.text.split("REACT_APP_CHECKOUT_API_URL: '")[-1].split("',")[0]
+    ld_proxy_domain = payment_page.text.split("REACT_APP_LD_PROXY_URL: '")[-1].split("',")[0]
+    spreedly_api_url = payment_page.text.split("REACT_APP_SPREEDLY_API_URL: '")[-1].split("',")[0]
+
+    # Sanity checks on the variables
+    assert 'paddle' in checkout_api_url
+    checkout_api_url = f'{checkout_api_url}/checkout/{checkout_id}'
+    if not checkout_api_url.endswith('/'):
+        checkout_api_url += '/'
+
+    assert 'paddle' in ld_proxy_domain
+
+    assert 'spreedly' in spreedly_api_url
+    assert spreedly_api_url.endswith('v1'), 'Unexpected version of the spreedly API, this test might not work correctly'
+    if not spreedly_api_url.endswith('/'):
+        spreedly_api_url += '/'
+
+    # prepare the customer info request
+    customer_info_url = parse.urljoin(checkout_api_url, 'customer-info')
+    customer_info = requests.post(
+        customer_info_url,
+        json={
+            'data': {
+                'email': email,
+                'audience_optin': False,
+                'country_code': 'PL',
+                'postcode': 12345
+            }
+        }
+    )
+    payment_methods = customer_info.json().get(
+        'data', {}
+    ).get(
+        'available_payment_methods', []
+    )
+
+    # fetch the CARD payment method
+    for payment_method in payment_methods:
+        if payment_method['type'] == 'CARD':
+            assert 'spreedly_options' in payment_method
+            assert 'spreedly_environment_key' in payment_method['spreedly_options']
```

Comment:
No need to check it here, since if it doesn't exist then it will fail later when doing
```python
spreedly_environment_key = payment_method['spreedly_options']['spreedly_environment_key']
```

Reply:
:+1: